core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -67,17 +67,16 @@ private[spark] class PythonRDD(
       envVars += ("SPARK_REUSE_WORKER" -> "1")
     }
     val worker: Socket = env.createPythonWorker(pythonExec, envVars.toMap)
+    // Whether is the worker released into idle pool
+    var released = false
```

Comment:
Do you think that this should be `@volatile`, since it's written from PythonRDD but read in a callback, which could possibly be executed in a different thread?  If you think it should be, I can just fix that up myself on merge.